File: testdata/jennies/builders/nullable_map_assignment/PHPBuilder/src/NullableMapAssignment/SomeStructBuilder.php
<?php

namespace Grafana\Foundation\NullableMapAssignment;

/**
 * @implements \Grafana\Foundation\Cog\Builder<\Grafana\Foundation\NullableMapAssignment\SomeStruct>
 */
class SomeStructBuilder implements \Grafana\Foundation\Cog\Builder
{
    protected \Grafana\Foundation\NullableMapAssignment\SomeStruct $internal;

    public function __construct()
    {
    	$this->internal = new \Grafana\Foundation\NullableMapAssignment\SomeStruct();
    }

    /**
     * Builds the object.
     * @return \Grafana\Foundation\NullableMapAssignment\SomeStruct
     */
    public function build()
    {
        return $this->internal;
    }

    /**
     * @param array<string, string> $config
     */
    public function config(array $config): static
    {
        $this->internal->config = $config;
    
        return $this;
    }

}

File: testdata/jennies/builders/nullable_map_assignment/JavaBuilder/nullable_map_assignment/SomeStructBuilder.java
package nullable_map_assignment;

import java.util.Map;

public class SomeStructBuilder implements cog.Builder<SomeStruct> {
    protected final SomeStruct internal;
    
    public SomeStructBuilder() {
        this.internal = new SomeStruct();
    }
    public SomeStructBuilder config(Map<String, String> config) {
        this.internal.config = config;
        return this;
    }
    public SomeStruct build() {
        return this.internal;
    }
}
